src/main/java/org/apache/mesos/offer/MesosResourcePool.java

```diff
@@ -67,6 +66,32 @@ public MesosResource consume(ResourceRequirement resReq) {
     return null;
   }
 
+  public MesosResource consume(DynamicPortRequirement dynamicPortRequirement) {
+    Value availableValue = unreservedMergedPool.get(dynamicPortRequirement.getName());
+
+    if (availableValue == null) {
+      return null;
+    }
+
+    // Choose first available port
+    if (availableValue.getRanges().getRangeCount() > 0) {
+      Value.Range range = availableValue.getRanges().getRange(0);
+      Resource resource = ResourceUtils.getUnreservedResource(
+              dynamicPortRequirement.getName(),
+              Value.newBuilder()
+                      .setType(Value.Type.RANGES)
+                      .setRanges(Value.Ranges.newBuilder()
+                              .addRange(Value.Range.newBuilder()
+                                      .setBegin(range.getBegin())
+                                      .setEnd(range.getBegin())))
```

Comment:
maybe leave a comment saying that it's intentionally using getBegin() here to just grab one port (included in above example)